src/rgw/rgw_sal_motr.cc

```diff
@@ -2215,7 +2215,34 @@ int MotrObject::remove_mobj_and_index_entry(
   if (ent.meta.size != 0) {
     if (ent.meta.category == RGWObjCategory::MultiMeta) {
       this->set_category(RGWObjCategory::MultiMeta);
-      rc = this->delete_part_objs(dpp, &size_rounded);
+      if (store->gc_enabled()) {
```

Comment:
we can take out this condition now so that it will be common for Simple and Multipart Objects. Just the tag generation will be different.
But we can handle this in separate PR as well

Reply:
Yes, this we will handle with separate PR.